frodo.js

```diff
@@ -6,70 +6,69 @@
  *
```

Comment:
SCRIPT-This new feature lacks unit tests. Could you add some to ensure everything works correctly?